core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineMysql.java

```diff
@@ -50,7 +50,52 @@ protected void createTableInternalExecuteAfterCreateTable(
   }
 
   @Override
-  protected RdbEngine getRdbEngine() {
+  boolean isDuplicateUserError(SQLException e) {
+    throw new UnsupportedOperationException();
```

Comment:
Yes, it's another possible design but I decided to throw exception here because:

- currently, `RdbEngineMysql.isDuplicateUserError()` is not (directly/indirectly) called from anywhere
- just `return false;` here may lead to future bugs if `RdbEngineMysql.isDuplicateUserError()` is called from somewhere
- so the perfect solution would be to check DuplicateUserError here but I didn't for now.